File: UI/eParking_mobile/lib/models/car.dart
class Car {
  final int id;
  final int brandId;
  final int colorId;
  final int userId;
  final String model;
  final String licensePlate;
  final String brandName;
  final String colorName;
  final bool isActive;
  final String? picture;

  Car({
    required this.id,
    required this.brandId,
    required this.colorId,
    required this.userId,
    required this.model,
    required this.licensePlate,
    required this.brandName,
    required this.colorName,
    required this.isActive,
    this.picture,
  });

  factory Car.fromJson(Map<String, dynamic> json) {
    return Car(
      id: json['id'] as int,
      brandId: json['brandId'] as int,
      colorId: json['colorId'] as int,
      userId: json['userId'] as int,
      model: json['model'] as String? ?? '',
      licensePlate: json['licensePlate'] as String? ?? '',
      brandName: json['brandName'] as String? ?? '',
      colorName: json['colorName'] as String? ?? '',
      isActive: json['isActive'] as bool? ?? true,
      picture: json['picture'] as String?,
    );
  }

  String get label => '$brandName $model ($licensePlate)';

  String get subtitle => '$colorName · $licensePlate';
}

File: eParking.Services/Database/Parking/Car.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace eParking.Services.Database.Parking
{
    public class Car
    {
        [Key]
        public int Id { get; set; }

        [Required]
        public int BrandId { get; set; }

        [ForeignKey(nameof(BrandId))]
        public Brand Brand { get; set; } = null!;

        [Required]
        public int ColorId { get; set; }

        [ForeignKey(nameof(ColorId))]
        public Color Color { get; set; } = null!;

        [Required]
        public int UserId { get; set; }

        [ForeignKey(nameof(UserId))]
        public MyAppUser User { get; set; } = null!;

        [Required]
        [MaxLength(100)]
        public string Model { get; set; } = string.Empty;

        [Required]
        [MaxLength(20)]
        public string LicensePlate { get; set; } = string.Empty;

        public int? YearOfManufacture { get; set; }

        public byte[]? Picture { get; set; }

        public bool IsActive { get; set; } = true;

        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

        public DateTime? UpdatedAt { get; set; }

        public ICollection<Reservation> Reservations { get; set; } = new List<Reservation>();
    }
}
